### PR TITLE
[2.25.x] Upgrade to Solr 8.9.0

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Back port of #6264, #6526, #6612.

#### Who is reviewing it? 

@dcruver 

#### Select relevant component teams: 

@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.

@mcalcote

#### How should this be tested?

Verify CI build passes

#### What are the relevant tickets?
Fixes: n/a

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

File: distribution/ddf-catalog/pom.xml [hunk 1]
```diff
@@ -156,7 +156,7 @@
                         <configuration>
                             <rules>
                                 <requireFilesSize>
-                                    <maxsize>310000000</maxsize>
+                                    <maxsize>311000000</maxsize>
```
Comment (mcalcote) [inline distribution/ddf-catalog/pom.xml:159]: Why the bump here?

**Review by mcalcote — APPROVED** [review]

File: distribution/ddf-catalog/pom.xml (hunk 1 above)
Comment (pklinef) [reply distribution/ddf-catalog/pom.xml:159]: DDF 2.25 was very close to max size before the upgrade. The Solr upgrade put it slightly over that limit. The Solr dependencies jar is 16.2 MB after the upgrade. The biggest dependencies in that jar are Solr, Lucene, and Zookeeper libraries. The main branch has had a lot removed from it so it did not require this.

**Pull request merged**